app/middleware/validation.middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { plainToClass } from 'class-transformer';
+import { validate, ValidationError } from 'class-validator';
+import * as express from 'express';
+import HttpException from '../exceptions/HttpException';
+
+function validationMiddleware<T>(type: any): express.RequestHandler {
+  return (req, res, next) => {
+    validate(plainToClass(type, req.body)).then((errors: ValidationError[]) => {
+      if (errors.length > 0) {
+        const message =
+          'Validation error(s): ' +
+          errors.map((error: ValidationError) => Object.values(error.constraints)).join(', ');
```

Comment:
I think type for `error` here is inferred so there is no need to explicitly type it